src/integrationTest/java/uk/gov/hmcts/darts/common/controller/CourthouseApiTest.java

```diff
@@ -0,0 +1,179 @@
+        MvcResult response = mockMvc.perform(requestBuilder).andExpect(status().isBadRequest()).andReturn();
+        assertEquals("{\"code\":\"400 BAD_REQUEST\",\"message\":\"courthouseName must not be null\"}", response.getResponse().getContentAsString());
+    }
+
+    /**
+     * Test utility method used to add courthouse to database.
+     *
+     * @param fileLocation location of file that contains courthouse to be added.
+     * @return response for successful add
+     */
+    private MvcResult makeRequestToAddCourthouseToDatabase(String fileLocation) throws Exception {
+        MockHttpServletRequestBuilder requestBuilder = post("/courthouses")
+            .contentType(MediaType.APPLICATION_JSON_VALUE)
+            .content(getContentsFromFile(fileLocation));
+        return mockMvc.perform(requestBuilder).andExpect(status().is2xxSuccessful()).andDo(print()).andReturn();
+    }
+
+
+    /**
+     * Test adds courthouse then amends it and then retrieves it to check amendment was made.
+     */
+    @Test
+    void courthousesPut() throws Exception {
+        MvcResult addResponse = makeRequestToAddCourthouseToDatabase(
+            REQUEST_BODY_HAVERFORDWEST_JSON);
+
+        Integer addedEntityId = JsonPath.read(addResponse.getResponse().getContentAsString(), "$.id");
+
+        String requestBody = getContentsFromFile("Tests/CourthousesTest/courthousesPutEndpoint/requestBodySwansea.json");
+        MockHttpServletRequestBuilder requestBuilder = put("/courthouses/{courthouse_id}", addedEntityId)
+            .contentType(MediaType.APPLICATION_JSON_VALUE)
+            .content(requestBody);
+        mockMvc.perform(requestBuilder).andExpect(status().isNoContent()).andReturn();
+
+        requestBuilder = get("/courthouses/{courthouse_id}", addedEntityId)
+            .contentType(MediaType.APPLICATION_JSON_VALUE);
+
+        mockMvc.perform(requestBuilder).andExpect(status().is2xxSuccessful())
+            .andExpect(jsonPath("$.courthouse_name", is("SWANSEA")))
+            .andExpect(jsonPath("$.code", is(457)))
+            .andExpect(jsonPath("$.created_date_time", is(notNullValue())))
+            .andExpect(jsonPath("$.last_modified_date_time", is(notNullValue())));
+    }
+
+    /*
+        Test adds a courthouse, deletes it and then tries to retrieve it but should get no content response.
+     */
+    @Test
+    void courthousesDelete() throws Exception {
+        makeRequestToAddCourthouseToDatabase(
+            REQUEST_BODY_HAVERFORDWEST_JSON);
+
+        MockHttpServletRequestBuilder requestBuilder = delete("/courthouses/{courthouse_id}", 1)
+            .contentType(MediaType.APPLICATION_JSON_VALUE);
+        mockMvc.perform(requestBuilder).andExpect(status().isNoContent()).andReturn();
+
+        requestBuilder = get("/courthouses/{courthouse_id}", 1)
+            .contentType(MediaType.APPLICATION_JSON_VALUE);
+        mockMvc.perform(requestBuilder).andExpect(status().isNoContent());
```

Comment:
Should this not be a 404 Not Found?

Reply:
`GET /courthouses/<id that does not exist>` should return 404. If `GET /court-houses/<id>` is called, that could also return 404, and I think that's ok. It's really an implementation issue rather than anything else.